File: tests/0561_arrayPartitionITest.cpp
#include "0561_arrayPartitionI.cpp"
#include "catch.hpp"

TEST_CASE("arrayPairSum")
{
    // setup
    std::vector<int> list{1, 4, 3, 2};
    Solution solution = Solution();

    REQUIRE(solution.arrayPairSum(list) == 4);

    list.clear();
    list.push_back(7);
    list.push_back(3);
    list.push_back(1);
    list.push_back(0);
    list.push_back(0);
    list.push_back(6);
    REQUIRE(solution.arrayPairSum(list) == 7);
}

File: tests/0561_array_partition_i_test.py
import importlib
import pytest

module = importlib.import_module("algorithms.0561_array_partition_i")


def test_valid_parentheses():
    s = module.Solution()
    assert s.arrayPairSum([1, 4, 3, 2]) == 4
    assert s.arrayPairSum([7, 3, 1, 0, 0, 6]) == 7
